include/camp/resource/cuda.hpp

```diff
@@ -259,6 +260,12 @@ namespace resources
       cudaStream_t get_stream() { return stream; }
       int get_device() { return device; }
 
+      bool compare (Cuda* c)
```

Comment:
Mostly this PR looks good, and like a nice addition, but the interface is slightly skewed between the type-erased and the strongly-typed classes. Would it be possible to either make the base class takes a pointer or, my preference, make each of these takes a reference so they match?  Having to sometimes do `.compare(&res)` and sometimes not seems unfortunate.

Reply:
I think recent changes addressed this, but let me know if something is still missing